sequencer/src/l1_client.rs

```diff
@@ -213,40 +235,28 @@ async fn update_loop(
 async fn get_finalized_block<P: JsonRpcClient>(
     rpc: &Provider<P>,
     tag: BlockNumber,
-    retry_delay: Duration,
-) -> Option<L1BlockInfo> {
-    // This cannot fail, retry until we succeed.
-    loop {
-        let block = match rpc.get_block(tag).await {
-            Ok(Some(block)) => block,
-            Ok(None) => {
-                // This can happen in rare cases where the L1 chain is very young and has not
-                // finalized a block yet. This is more common in testing and demo environments. In
-                // any case, we proceed with a null L1 block rather than wait for the L1 to finalize
-                // a block, which can take a long time.
-                tracing::warn!("no finalized block yet");
-                return None;
-            }
-            Err(err) => {
-                tracing::error!("failed to get finalized block from L1 provider, retrying ({err})");
-                sleep(retry_delay).await;
-                continue;
-            }
-        };
-
-        // The block number always exists unless the block is pending. The finalized block cannot be
-        // pending, unless there has been a catastrophic reorg of the finalized prefix of the L1
-        // chain, so it is OK to panic if this happens.
-        let number = block.number.expect("finalized block has no number");
-        // Same for the hash.
-        let hash = block.hash.expect("finalized block has no hash");
-
-        break Some(L1BlockInfo {
-            number: number.as_u64(),
-            timestamp: block.timestamp,
-            hash,
-        });
-    }
+) -> Result<Option<L1BlockInfo>, ProviderError> {
+    let Some(block) = rpc.get_block(tag).await? else {
+        // This can happen in rare cases where the L1 chain is very young and has not finalized a
+        // block yet. This is more common in testing and demo environments. In any case, we proceed
+        // with a null L1 block rather than wait for the L1 to finalize a block, which can take a
+        // long time.
+        tracing::warn!("no finalized block yet");
+        return Ok(None);
+    };
+
+    // The block number always exists unless the block is pending. The finalized block cannot be
+    // pending, unless there has been a catastrophic reorg of the finalized prefix of the L1
+    // chain, so it is OK to panic if this happens.
+    let number = block.number.expect("finalized block has no number");
```

Comment:
Since this function returns a Result now, can we remove these panics for resiliency's sake?

Reply:
Sure, good idea